packages/ovee/src/dom/EventDelegate.ts

```diff
@@ -3,143 +3,153 @@ import Component from 'src/core/Component';
-    off(
-        _events: string,
-        _target: Callback<Context> | string | Element,
-        _selector?: Callback<Context> | string,
-        _callback?: Callback<Context>
-    ): void {
-        const {
-            events, target, selector, callback
-        } = this.getCastedParams(_events, _target, _selector, _callback);
-
-        events.split(' ').forEach((event) => {
-            this.listeners.forEach((item, index) => {
-                if (item.event === event && item.target === target
-                    && item.selector === selector && item.callback === callback) {
-                    target.removeEventListener(event, item.handler);
-                    this.listeners.splice(index, 1);
-                }
-            });
-        });
-    }
-
-    private getCastedParams(
-        events: string,
-        target: Callback<Context> | string | Element,
-        selector?: Callback<Context> | string,
-        callback?: Callback<Context>
-    ) {
-        const out: Partial<CastedParams<Context>> = { events, callback };
-
-        if (callback === undefined) {
-            if (selector === undefined) {
-                out.callback = target as Callback<Context>;
-                out.target = this.targetElement;
-            } else if (typeof target === 'string') {
-                out.callback = selector as Callback<Context>;
-                out.selector = target;
-                out.target = this.targetElement;
-            } else {
-                out.callback = selector as Callback<Context>;
-                out.target = target as Element;
-            }
-        } else {
-            out.target = target as Element;
-            out.selector = selector as string;
-        }
-
-        return out as CastedParams<Context>;
-    }
-
-    emit<D = any>(eventDesc: EventDesc, detail?: D): void {
-        emitEvent(this.targetElement, eventDesc, detail);
-    }
-
-    destroy(): void {
-        this.listeners = this.listeners.filter((item) => {
-            item.target.removeEventListener(item.event, item.handler);
-            return false;
-        });
-    }
+	listeners: Listener<Context>[] = [];
+	targetElement: Element;
+	context: Context;
+
+	constructor(targetElement: Element, context: Context) {
+		this.targetElement = targetElement;
+		this.context = context;
+	}
+
+	on(events: string, callback: Callback<Context>): void;
+	on(events: string, selector: string, callback: Callback<Context>): void;
+	on(events: string, target: Element, callback: Callback<Context>): void;
+	on(events: string, target: Element, selector: string, callback: Callback<Context>): void;
+	on(
+		_events: string,
+		_target: Callback<Context> | string | Element,
+		_selector?: Callback<Context> | string,
+		_callback?: Callback<Context>
+	): void {
+		const { events, target, selector, callback } = this.getCastedParams(
+			_events,
+			_target,
+			_selector,
+			_callback
+		);
+
+		events.split(' ').forEach(event => {
```

Comment:
would be good if we handle the case of eg a double space or other unexpected characters

Reply:
good idea

as removing repeated whitespaces is easy, what special characters should we avoid? 🤔

Reply:
I've reviewed the docs and as far as I understand, the `Event` constructor does not constrain you from using any character as the event name. So probably we should just handle double spaces there, and trust in our users to keep things clean.